packages/@internationalized/date/src/string.ts

```diff
@@ -22,6 +22,8 @@ const DATE_RE = /^(\d{4})-(\d{2})-(\d{2})$/;
 const DATE_TIME_RE = /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}))?(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?$/;
 const ZONED_DATE_TIME_RE = /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}))?(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?(?:([+-]\d{2})(?::(\d{2}))?)?\[(.*?)\]$/;
 const ABSOLUTE_RE = /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}))?(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?(?:(?:([+-]\d{2})(?::(\d{2}))?)|Z)$/;
+const DATE_TIME_DURATION_RE =
+    /^(?<negative>-)?P((?<years>-?\d*[.,]?\d+)Y)?((?<months>-?\d*[.,]?\d+)M)?((?<weeks>-?\d*[.,]?\d+)W)?((?<days>-?\d*[.,]?\d+)D)?(T((?<hours>-?\d*[.,]?\d+)H)?((?<minutes>-?\d*[.,]?\d+)M)?((?<seconds>-?\d*[.,]?\d+)S)?)?/;
```

Comment:
Maybe this regex should be more strict? I guess years, months, and days probably shouldn't accept fractional values, and not sure about commas.

Reply:
ISO8601 Duration Strings can contain fractional values for years, months, and days and decimal values can be expressed with commas or periods, hence the regex accepting those values.

Reply:
According to the TC39 Temporal specification, years, months, and days only support whole number values whereas hours, minutes, and seconds support fractions. https://tc39.es/proposal-temporal/#prod-Duration

You can also see in the official tests some examples of invalid duration strings: https://github.com/tc39/test262/blob/main/test/built-ins/Temporal/Duration/from/argument-string-invalid.js We should make sure that we pass all of those. The goal is eventually to replace the internals of `@internationalized/date` with temporal once it is implemented natively in browsers, so we want to ensure the behavior is compatible as much as possible.

Reply:
Updated to align with the TC39 Temporal specs you shared - thanks Devon!